pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/TextIndexConfig.java

```diff
@@ -61,7 +65,9 @@ public TextIndexConfig(@JsonProperty("disabled") Boolean disabled, @JsonProperty
       @JsonProperty("luceneUseCompoundFile") Boolean luceneUseCompoundFile,
       @JsonProperty("luceneMaxBufferSizeMB") Integer luceneMaxBufferSizeMB,
       @JsonProperty("luceneAnalyzerClass") String luceneAnalyzerClass,
-      @JsonProperty("enablePrefixSuffixMatchingInPhraseQueries") Boolean enablePrefixSuffixMatchingInPhraseQueries) {
+      @JsonProperty("enablePrefixSuffixMatchingInPhraseQueries") Boolean enablePrefixSuffixMatchingInPhraseQueries,
+      @JsonProperty("reuseMutableIndex") Boolean reuseMutableIndex,
+      @JsonProperty("enableNRTCachingDirectory") Boolean enableNRTCachingDirectory) {
```

Comment:
We could use a config `luceneNRTCachingDirectoryBufferSize` instead of reusing `luceneMaxBufferSizeMB` for the NRTCachingDirectory's size, which has the benefit of allowing users to disable the functionality w/ size 0 as this config does, and also adjust it independently of IndexWriter buffer

Reply:
sounds good. updating.